pyvista/core/filters/data_set.py

```diff
@@ -384,7 +401,10 @@ def _validate_vector(vector, name):
                 vector = _validation.validate_array3(vector, dtype_out=float, name=name)
             return vector
 
-        axes, std = pyvista.principal_axes(self.points, return_std=True)
+        # Get points and compute principal axes
+        input_mesh = self.cell_centers() if cell_centers else self
+        points = input_mesh.merge_points().points if merge_points else input_mesh.points
+        axes, std = pyvista.principal_axes(points, return_std=True)
```

Comment:
It seems natural to me that use `cell_centers` defaults in my mind. I leave the final decision to you. I would like to add that since this method is not yet released, it is not a problem to change the default behavior.

Reply:
Yes that does seem natural. It should also implicitly remove any real need to use `merge_points=True` for these cases since that's meant to address duplicates caused 
by having independent points for the cells. 

I'll make this option `None` by default and use cells if available, use points otherwise (e.g. PointSet)